### ISSUE TITLE
Compilation/linking error: `ld: symbol(s) not found for architecture x86_64`

### ISSUE THREAD

**Issue by espenhgn** [nest/nest-simulator#1678]
**Describe the bug**
I'm seeing this error while compiling the current master (at ba76461b140e17d41a7061d2e6831dde36d75539),
which does not appear to be a duplicate of any closed issues I could find:
```
$ make -j4
...
[ 96%] Building CXX object nest/CMakeFiles/nest_lib.dir/neststartup.cpp.o
[ 97%] Building CXX object nest/CMakeFiles/nest_lib.dir/sli_neuron.cpp.o
[ 97%] Linking CXX executable nest
[ 97%] Built target nest
Scanning dependencies of target run_all_cpptests
[ 98%] Building CXX object testsuite/cpptests/CMakeFiles/run_all_cpptests.dir/run_all.cpp.o
[ 98%] Linking CXX shared library libnest.dylib
ld: warning: -pie being ignored. It is only used when linking a main executable
Undefined symbols for architecture x86_64:
  "_PyCapsule_GetName", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
  "_PyCapsule_GetPointer", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
  "_PyCapsule_IsValid", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
  "_PyDict_GetItemString", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
  "_PyErr_Format", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
      __Pyx_ImportType(_object*, char const*, char const*, unsigned long, __Pyx_ImportType_CheckSize) in neststartup.cpp.o
  "_PyErr_WarnEx", referenced from:
      __Pyx_ImportType(_object*, char const*, char const*, unsigned long, __Pyx_ImportType_CheckSize) in neststartup.cpp.o
  "_PyExc_ImportError", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
  "_PyExc_TypeError", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
      __Pyx_ImportType(_object*, char const*, char const*, unsigned long, __Pyx_ImportType_CheckSize) in neststartup.cpp.o
  "_PyExc_ValueError", referenced from:
      __Pyx_ImportType(_object*, char const*, char const*, unsigned long, __Pyx_ImportType_CheckSize) in neststartup.cpp.o
  "_PyImport_ImportModule", referenced from:
      set_communicator(_object*) in neststartup.cpp.o
  "_PyModule_GetName", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
  "_PyOS_snprintf", referenced from:
      __Pyx_ImportType(_object*, char const*, char const*, unsigned long, __Pyx_ImportType_CheckSize) in neststartup.cpp.o
  "_PyObject_GetAttrString", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
      __Pyx_ImportType(_object*, char const*, char const*, unsigned long, __Pyx_ImportType_CheckSize) in neststartup.cpp.o
  "_PyType_IsSubtype", referenced from:
      set_communicator(_object*) in neststartup.cpp.o
  "__Py_Dealloc", referenced from:
      set_communicator(_object*) in neststartup.cpp.o
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
      __Pyx_ImportType(_object*, char const*, char const*, unsigned long, __Pyx_ImportType_CheckSize) in neststartup.cpp.o
ld: symbol(s) not found for architecture x86_64
clang-9: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [nest/libnest.dylib] Error 1
make[1]: *** [nest/CMakeFiles/nest_lib.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**To Reproduce**
1. Conda env file:
```
name: nest
channels:
  - conda-forge
  - defaults
dependencies:
  - python=3
  - openmpi-mpicc
  - openmpi-mpicxx
  - ipython
  - boost
  - libtool
  - readline
  - mpi4py
  - openmp
  - gsl
  - cython
  - doxygen
  - graphviz  # for dot
  - pkg-config
```
2. packages:
```
$ conda list
# packages in environment at /Users/ehagen/anaconda3/envs/nest:
#
# Name                    Version                   Build  Channel
appnope                   0.1.0           py38h32f6830_1001    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
boost                     1.73.0           py38hd74bcbf_0    conda-forge
boost-cpp                 1.73.0               hef959ae_1    conda-forge
bzip2                     1.0.8                h0b31af3_2    conda-forge
ca-certificates           2020.6.20            hecda079_0    conda-forge
cctools                   927.0.2              h5ba7a2e_4    conda-forge
certifi                   2020.6.20        py38h32f6830_0    conda-forge
clang                     9.0.1           default_hf57f61e_0    conda-forge
clang-tools               9.0.1           default_hf57f61e_0    conda-forge
clang_osx-64              9.0.1                h05bbb7f_0    conda-forge
clangxx                   9.0.1           default_hf57f61e_0    conda-forge
clangxx_osx-64            9.0.1                h05bbb7f_2    conda-forge
compiler-rt               9.0.1                h6a512c6_3    conda-forge
compiler-rt_osx-64        9.0.1                h99342c6_3    conda-forge
cython                    0.29.20          py38hc84c608_0    conda-forge
decorator                 4.4.2                      py_0    conda-forge
doxygen                   1.8.18               h72f44dd_0    conda-forge
freetype                  2.10.2               h8da9a1a_0    conda-forge
graphviz                  2.42.3               h98dfb87_0    conda-forge
gsl                       2.6                  ha2d443c_0    conda-forge
icu                       67.1                 h4a8c4bd_0    conda-forge
ipython                   7.16.1           py38h23f93f0_0    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
jedi                      0.17.1           py38h32f6830_0    conda-forge
jpeg                      9d                   h0b31af3_0    conda-forge
ld64                      450.3                h3c32e8a_4    conda-forge
libblas                   3.8.0               14_openblas    conda-forge
libcblas                  3.8.0               14_openblas    conda-forge
libcxx                    10.0.0               h1af66ff_2    conda-forge
libffi                    3.2.1             h4a8c4bd_1007    conda-forge
libgfortran               4.0.0                         2    conda-forge
libiconv                  1.15              h0b31af3_1006    conda-forge
liblapack                 3.8.0               14_openblas    conda-forge
libllvm9                  9.0.1                h7475705_1    conda-forge
libopenblas               0.3.7                h3d69b6c_4    conda-forge
libpng                    1.6.37               hbbe82c9_1    conda-forge
libtiff                   4.1.0                h2ae36a8_6    conda-forge
libtool                   2.4.6             h1de35cc_1002    conda-forge
libwebp-base              1.1.0                h0b31af3_3    conda-forge
llvm-openmp               8.0.1                h770b8ee_0    conda-forge
lz4-c                     1.9.2                h4a8c4bd_1    conda-forge
mpi                       1.0                     openmpi    conda-forge
mpi4py                    3.0.3            py38h5dec745_1    conda-forge
ncurses                   6.1               h0a44026_1002    conda-forge
numpy                     1.18.5           py38h1f821a2_0    conda-forge
openmp                    8.0.1                         0    conda-forge
openmpi                   4.0.4                hed52333_0    conda-forge
openmpi-mpicc             4.0.4                ha1c9252_0    conda-forge
openmpi-mpicxx            4.0.4                h962d5db_0    conda-forge
openssl                   1.1.1g               h0b31af3_0    conda-forge
parso                     0.7.0              pyh9f0ad1d_0    conda-forge
pexpect                   4.8.0            py38h32f6830_1    conda-forge
pickleshare               0.7.5           py38h32f6830_1001    conda-forge
pip                       20.1.1                     py_1    conda-forge
pkg-config                0.29.2            hdc84dc8_1006    conda-forge
prompt-toolkit            3.0.5                      py_1    conda-forge
ptyprocess                0.6.0                   py_1001    conda-forge
pygments                  2.6.1                      py_0    conda-forge
python                    3.8.3           cpython_hd5f0129_0    conda-forge
python_abi                3.8                      1_cp38    conda-forge
readline                  8.0                  hcfe32e1_0    conda-forge
setuptools                49.1.0           py38h32f6830_0    conda-forge
six                       1.15.0             pyh9f0ad1d_0    conda-forge
sqlite                    3.32.3               h93121df_0    conda-forge
tapi                      1000.10.8            ha1b3eb9_4    conda-forge
tk                        8.6.10               hbbe82c9_0    conda-forge
traitlets                 4.3.3            py38h32f6830_1    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_0    conda-forge
wheel                     0.34.2                     py_1    conda-forge
xz                        5.2.5                h0b31af3_0    conda-forge
zlib                      1.2.11            h0b31af3_1006    conda-forge
zstd                      1.4.4                h4b3e974_3    conda-forge
```
3. misc.
```
$ conda info
     active environment : nest
    active env location : /Users/ehagen/anaconda3/envs/nest
            shell level : 1
       user config file : /Users/ehagen/.condarc
 populated config files : /Users/ehagen/.condarc
          conda version : 4.8.3
    conda-build version : 3.19.1
         python version : 3.7.6.final.0
       virtual packages : __osx=10.14.6
       base environment : /Users/ehagen/anaconda3  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/ehagen/anaconda3/pkgs
                          /Users/ehagen/.conda/pkgs
       envs directories : /Users/ehagen/anaconda3/envs
                          /Users/ehagen/.conda/envs
               platform : osx-64
             user-agent : conda/4.8.3 requests/2.23.0 CPython/3.7.6 Darwin/18.7.0 OSX/10.14.6
                UID:GID : 505:20
             netrc file : None
           offline mode : False
```
4. Build script:
```
# create conda env
conda env create -f nest_environment.yml
conda activate nest

# conda environment path
export CONDAENV=$HOME/anaconda3/envs/nest

# compilers
export CC=$CONDAENV/bin/mpicc
export CXX=$CONDAENV/bin/mpicxx

# path variables
    export NESTBLD=$HOME/Repositories/nest-simulator-bld
    export NESTSOURCES=$HOME/Repositories/nest-simulator
    export NESTPREFIX=$CONDAENV

# create build directory
mkdir $NESTBLDs
cd $NESTBLD
rm -r *

# configure NEST
cmake -DCMAKE_INSTALL_PREFIX:PATH=$NESTPREFIX \
      -Dwith-mpi=ON \
      -Dwith-python=ON \
      -DLTDL_ROOT_DIR=$CONDAENV \
      -Dwith-gsl=$CONDAENV \
      -Dwith-openmp=ON \
      -Dwith-readline=$CONDAENV \
      -DCMAKE_C_COMPILER=$CC \
      -DCMAKE_CXX_COMPILER=$CXX \
      $NESTSOURCES
make
```
4. cmake summary
```
Build type          : 
Target System       : Darwin
Cross Compiling     : FALSE
C compiler          : Clang 9.0.1 (/Users/ehagen/anaconda3/envs/nest/bin/mpicc)
C compiler flags    : -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/ehagen/anaconda3/envs/nest/include -O2 -Wall -fopenmp=libomp    
C++ compiler        : Clang 9.0.1 (/Users/ehagen/anaconda3/envs/nest/bin/mpicxx)
C++ compiler flags  : -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -std=c++14 -fmessage-length=0 -isystem /Users/ehagen/anaconda3/envs/hybridlfpy/include -std=c++11 -O2 -Wall -fopenmp=libomp  
Build dynamic       : ON

Built-in modules    : models;precise;topology
User modules        : None
Python bindings     : Yes (Python 3.8.3: /Users/ehagen/anaconda3/envs/nest/bin/python3)
       Includes     : /Users/ehagen/anaconda3/envs/nest/include/python3.8
       Libraries    : /Users/ehagen/anaconda3/envs/nest/lib/libpython3.8.dylib

Cython bindings     : Yes (Cython 0.29.20: /Users/ehagen/anaconda3/envs/nest/bin/cython)
MPI4PY header	    : Yes (/Users/ehagen/anaconda3/envs/nest/lib/python3.8/site-packages/mpi4py/include)
Use threading       : Yes (OpenMP: -fopenmp=libomp)
Use GSL             : Yes (GSL 2.6)
    Includes        : /Users/ehagen/anaconda3/envs/nest/include
    Libraries       : /Users/ehagen/anaconda3/envs/nest/lib/libgsl.dylib;/Users/ehagen/anaconda3/envs/nest/lib/libgslcblas.dylib

Use Readline        : Yes (GNU Readline 8.0)
    Includes        : /Users/ehagen/anaconda3/envs/nest/include
    Libraries       : /Users/ehagen/anaconda3/envs/nest/lib/libreadline.dylib;/Users/ehagen/anaconda3/envs/nest/lib/libncurses.dylib

Use libltdl         : Yes (LTDL 2.4.6)
    Includes        : /Users/ehagen/anaconda3/envs/nest/include
    Libraries       : /Users/ehagen/anaconda3/envs/nest/lib/libltdl.dylib

Use doxygen         : Yes (/Users/ehagen/anaconda3/envs/nest/bin/doxygen)
                    : target `doc` available
    `dot` available : Yes (/Users/ehagen/anaconda3/envs/nest/bin/dot)
                    : target `fulldoc` available
Use MPI             : Yes (MPI: /Users/ehagen/anaconda3/envs/nest/bin/mpicxx)
    FLAGS           : 
    Includes        : 
    Link Flags      : 
    Libraries       : 

Use MUSIC           : No
Use libneurosim     : No
Use Boost           : Yes (Boost 1.73.0)
    Includes        : /Users/ehagen/anaconda3/envs/nest/include
    Libraries       : 

Use recording backend Arbor   : No
Use SIONlib         : No
```

**Expected behavior**
A successful build. 

**Desktop/Environment (please complete the following information):**
 - OS: macOS 10.14.6
 - Python-Version: 3.8
 - NEST-Version: ba76461b140e17d41a7061d2e6831dde36d75539
 - Installation: See above

**Comment by espenhgn** [comment]
Did some additional testing, and the same error occurs only with `-Dwith-mpi=ON` after disabling everything else but python (and doxygen), and using the `DMPI_C/CXX_COMPILER` options as suggested in INSTALL:
```
export CC=$CONDAENV/bin/clang
export CXX=$CONDAENV/bin/clang++
export MPICC=$CONDAENV/bin/mpicc
export MPICXX=$CONDAENV/bin/mpicxx

cmake -DCMAKE_INSTALL_PREFIX:PATH=$NESTPREFIX \
      -Dwith-mpi=ON \
      -Dwith-python=ON \
      -Dwith-ltdl=OFF \
      -Dwith-gsl=OFF \
      -Dwith-openmp=OFF \
      -Dwith-readline=OFF \
      -Dwith-boost=OFF \
      -DMPI_C_COMPILER=$MPICC \
      -DMPI_CXX_COMPILER=$MPICXX \
      -DCMAKE_C_COMPILER=$CC \
      -DCMAKE_CXX_COMPILER=$CXX \
      $NESTSOURCES
```
config. summary, which suggests that all MPI includes etc. are found:
```
Build type          : 
Target System       : Darwin
Cross Compiling     : FALSE
C compiler          : Clang 9.0.1 (/Users/ehagen/anaconda3/envs/nest/bin/clang)
C compiler flags    : -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/ehagen/anaconda3/envs/nest/include -O2 -Wall    
C++ compiler        : Clang 9.0.1 (/Users/ehagen/anaconda3/envs/nest/bin/clang++)
C++ compiler flags  : -march=core2 -mtune=haswell -mssse3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -std=c++14 -fmessage-length=0 -isystem /Users/ehagen/anaconda3/envs/nest/include -std=c++11 -O2 -Wall  
Build dynamic       : ON

Built-in modules    : models;precise;topology
User modules        : None
Python bindings     : Yes (Python 3.8.3: /Users/ehagen/anaconda3/envs/nest/bin/python3)
       Includes     : /Users/ehagen/anaconda3/envs/nest/include/python3.8
       Libraries    : /Users/ehagen/anaconda3/envs/nest/lib/libpython3.8.dylib

Cython bindings     : Yes (Cython 0.29.20: /Users/ehagen/anaconda3/envs/nest/bin/cython)
MPI4PY header	    : Yes (/Users/ehagen/anaconda3/envs/nest/lib/python3.8/site-packages/mpi4py/include)
Use threading       : No
Use GSL             : No
Use Readline        : No
Use libltdl         : No
Use doxygen         : Yes (/Users/ehagen/anaconda3/envs/nest/bin/doxygen)
                    : target `doc` available
    `dot` available : Yes (/Users/ehagen/anaconda3/envs/nest/bin/dot)
                    : target `fulldoc` available
Use MPI             : Yes (MPI: /Users/ehagen/anaconda3/envs/nest/bin/mpicxx)
    FLAGS           : 
    Includes        : /Users/ehagen/anaconda3/envs/nest/include
    Link Flags      : -Wl,-rpath,/Users/ehagen/anaconda3/envs/nest/lib
    Libraries       : /Users/ehagen/anaconda3/envs/nest/lib/libmpi.dylib

Use MUSIC           : No
Use libneurosim     : No
Use Boost           : No
Use recording backend Arbor   : No
Use SIONlib         : No
```

**Comment by gtrensch** [comment]
@espenhgn Thanks for reporting this! We will look for OSX expertise on this.

**Comment by heplesser** [comment]
@espenhgn I can reproduce your error when using you conda environment, but all works fine when building NEST using GCC from Homebrew. I will explore more. Something is probably going wrong about the conditional PyNEST inclusion in `neststartup`.

**Comment by espenhgn** [comment]
Hi @heplesser, thanks for confirming. I'm not using homebrew (for reasons discussed for example here: https://saagarjha.com/blog/2019/04/26/thoughts-on-macos-package-managers/) but macports which should be self contained. A similar compile error results also with GCC, python3.8 etc. provided by macports, so the issue is not only occurring with conda:
```
# compilers
export CC=/opt/local/bin/gcc-mp-10
export CXX=/opt/local/bin/g++-mp-10
export CC=/opt/local/bin/mpicc-openmpi-mp
export CXX=/opt/local/bin/mpicxx-openmpi-mp

# path variables
export NESTBLD=$HOME/Repositories/nest-simulator-bld
export NESTSOURCES=$HOME/Repositories/nest-simulator
export NESTPREFIX=$HOME/.local/nest_py38

cd $NESTLD

cmake -DCMAKE_INSTALL_PREFIX:PATH=$NESTPREFIX \
      -Dwith-mpi=ON \
      -Dwith-python=ON \
      -Dwith-ltdl=OFF \
      -Dwith-gsl=OFF \
      -Dwith-openmp=OFF \
      -Dwith-readline=OFF \
      -Dwith-boost=OFF \
      -DMPI_C_COMPILER=$MPICC \
      -DMPI_CXX_COMPILER=$MPICXX \
      -DCMAKE_C_COMPILER=$CC \
      -DCMAKE_CXX_COMPILER=$CXX \
      $NESTSOURCES
```
Config summary
```
--------------------------------------------------------------------------------
NEST Configuration Summary
--------------------------------------------------------------------------------

Build type          : 
Target System       : Darwin
Cross Compiling     : FALSE
C compiler          : AppleClang 10.0.1.10010046 (/opt/local/bin/mpicc-openmpi-mp)
C compiler flags    :  -O2 -Wall    
C++ compiler        : AppleClang 10.0.1.10010046 (/opt/local/bin/mpicxx-openmpi-mp)
C++ compiler flags  :  -std=c++11 -O2 -Wall  
Build dynamic       : ON

Built-in modules    : models;precise;topology
User modules        : None
Python bindings     : Yes (Python 3.8.3: /opt/local/bin/python)
       Includes     : /opt/local/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8
       Libraries    : /opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/libpython3.8.dylib

Cython bindings     : Yes (Cython 0.29.20: /opt/local/bin/cython)
MPI4PY header	    : Yes (/opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/mpi4py/include)
Use threading       : No
Use GSL             : No
Use Readline        : No
Use libltdl         : No
Use doxygen         : Yes (/opt/local/bin/doxygen)
                    : target `doc` available
    `dot` available : Yes (/opt/local/bin/dot)
                    : target `fulldoc` available
Use MPI             : Yes (MPI: /opt/local/bin/mpicxx-openmpi-mp)
    FLAGS           : 
    Includes        : 
    Link Flags      : 
    Libraries       : 

Use MUSIC           : No
Use libneurosim     : No
Use Boost           : No
Use recording backend Arbor   : No
Use SIONlib         : No

--------------------------------------------------------------------------------

The NEST executable will be installed to:
  /Users/ehagen/.local/nest_py38/bin/

NEST dynamic libraries and user modules will be installed to:
  /Users/ehagen/.local/nest_py38/lib/nest/

Documentation and examples will be installed to:
  /Users/ehagen/.local/nest_py38/share/doc/nest/

PyNEST will be installed to:
    /Users/ehagen/.local/nest_py38/lib/python3.8/site-packages

To set necessary environment variables, add the following line
to your ~/.bashrc :
  source /Users/ehagen/.local/nest_py38/bin/nest_vars.sh

--------------------------------------------------------------------------------
```
make error:
```
...
[ 98%] Linking CXX shared library libnest.dylib
Undefined symbols for architecture x86_64:
  "_PyCapsule_GetName", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
  "_PyCapsule_GetPointer", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
  "_PyCapsule_IsValid", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
  "_PyDict_GetItemString", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
  "_PyErr_Format", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
      __Pyx_ImportType(_object*, char const*, char const*, unsigned long, __Pyx_ImportType_CheckSize) in neststartup.cpp.o
  "_PyErr_WarnEx", referenced from:
      __Pyx_ImportType(_object*, char const*, char const*, unsigned long, __Pyx_ImportType_CheckSize) in neststartup.cpp.o
  "_PyExc_ImportError", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
  "_PyExc_TypeError", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
      __Pyx_ImportType(_object*, char const*, char const*, unsigned long, __Pyx_ImportType_CheckSize) in neststartup.cpp.o
  "_PyExc_ValueError", referenced from:
      __Pyx_ImportType(_object*, char const*, char const*, unsigned long, __Pyx_ImportType_CheckSize) in neststartup.cpp.o
  "_PyImport_ImportModule", referenced from:
      set_communicator(_object*) in neststartup.cpp.o
  "_PyModule_GetName", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
  "_PyOS_snprintf", referenced from:
      __Pyx_ImportType(_object*, char const*, char const*, unsigned long, __Pyx_ImportType_CheckSize) in neststartup.cpp.o
  "_PyObject_GetAttrString", referenced from:
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
      __Pyx_ImportType(_object*, char const*, char const*, unsigned long, __Pyx_ImportType_CheckSize) in neststartup.cpp.o
  "_PyType_IsSubtype", referenced from:
      set_communicator(_object*) in neststartup.cpp.o
  "__Py_Dealloc", referenced from:
      set_communicator(_object*) in neststartup.cpp.o
      __Pyx_ImportFunction(_object*, char const*, void (**)(), char const*) in neststartup.cpp.o
      __Pyx_ImportType(_object*, char const*, char const*, unsigned long, __Pyx_ImportType_CheckSize) in neststartup.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [nest/libnest.dylib] Error 1
make[1]: *** [nest/CMakeFiles/nest_lib.dir/all] Error 2
make: *** [all] Error 2
```
Let me know if I should try something else.

**Comment by heplesser** [comment]
@espenhgn Thanks for checking! I have now managed to narrow it down further. The problem is introduced by MPI4Py (which I did not have available in my Homebrew environment). If MPI4Py is available, then `neststartup.cpp` calls functions provided by MPI4Py, but `libnest.dylib` is not linked against a suitable mpi4py library. I seems one also needs to link against `libpython3.8.dylib`.

Does this/how does this work under Linux: Is linking against `libpython` not needed when building `libnest` under Linux or is the library explicitly provided under Linux but not under macOS?

I am working on a PR.

-
